sympy/solvers/solvers.py

```diff
@@ -849,13 +853,23 @@ def solve(f, *symbols, **flags):
     def _sympified_list(w):
         return list(map(sympify, w if iterable(w) else [w]))
     bare_f = not iterable(f)
+
+    # check flag usage for particular/quick which should only be used
+    # with systems of equations
+    if flags.get('quick', None) is not None:
+        flags['particular'] = True  # quick trumps particular
+    if flags.get('particular', False) and bare_f:
+        raise ValueError(filldedent("""
+            The 'particular/quick' flag is usually used with systems of
+            equations. Either pass your equation as a list or
+            consider using a solver like `diophantine` if you are
+            looking for a solution in integers."""))
```

Comment:
This seems like an unnecessary restriction. If we can have a system then why can't we have a system of size 1?

Reply:
Ideally there would be no difference between passing `[eq]` and `eq`. The fact that those aren't equivalent here is part of what makes solve such a mess and also is one of the things that caused difficulty when I tried to clean things up. A single equation given as `eq` should always just be treated as being equivalent to a system of 1 equation like `[eq]`.

Reply:
Maybe I'm misunderstanding how this works but I thought that the first argument is always the system of equations: it can be a single equation or a list of equations. The second argument is always either a list of unknowns or the first of the unknowns.